### PR TITLE
[core][test] Dynamic imports should be tested w/o conftest 

### PR DESCRIPTION
**Pull request by rickyyx**
## Why are these changes needed?

When running the tests like `test_dynamic_subpackage_import` in pytest, conftest gets loaded implicitly, which brings in many other ray's dependencies in the test. The test currently doesn't test correctly the ray dependency, might cause false negative. 

Running the intended test script in a subprocess should prevent the implicit imports. 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/tests/test_top_level_api.py [hunk 1]
```diff
@@ -56,45 +66,68 @@ def test_non_ray_modules():
...
-    ray.autoscaler
-    # Check that the package is cached.
-    assert "ray.autoscaler" in sys.modules
+# ray.data
+assert "ray.autoscaler" not in sys.modules
+ray.autoscaler
+# Check that the package is cached.
+assert "ray.autoscaler" in sys.modules
+"""
+    _run_script_in_subprocess(script)
```
Comment (jjyao) [inline python/ray/tests/test_top_level_api.py:91]: You can call `run_string_as_driver`?

Comment (rickyyx) [reply python/ray/tests/test_top_level_api.py:91]: oh i somehow thought it created a ray session - but yes, u r right.

**Review by rkooo567 — APPROVED** [review]
oh wow good catch. how did you find it

**Comment by rickyyx** [comment]
> oh wow good catch. how did you find it

i had a test that broke it but I think it's this test issue.

**Pull request merged**